pkg/apis/core/validation/validation.go

```diff
@@ -2767,6 +2785,10 @@ func validateContainers(containers []core.Container, isInitContainers bool, volu
 		allErrs = append(allErrs, ValidateSecurityContext(ctr.SecurityContext, idxPath.Child("securityContext"))...)
 	}
 
+	if !isInitContainers {
+		allErrs = append(allErrs, validateHasNonSidecar(containers, fldPath)...)
+	}
+
 	if isInitContainers {
```

Comment:
since in alpha feature we have decided not to add sidecar container among init containers, shall we validate this as well?

Reply:
Currently init containers don't support the Lifecycle field at all and they already have a validation for that. We could also add a check to make sure they don't have lifecycle type, not sure if necessary? or maybe a good idea to catch a regression?

Reply:
Validation already errors when lifecycle is set at all for init containers.